htlcswitch/switch.go

```diff
@@ -101,6 +100,18 @@ const (
 	CloseBreach
 )
 
+// PendingUpdate describes the pending state of a closing channel.
+type PendingUpdate struct {
+	Txid        []byte
+	OutputIndex uint32
+}
+
+// ChannelCloseUpdate contains the outcome of the close channel operation.
```

Comment:
FWIW, we've added two new structs to this package in this PR, yet they aren't referenced at all in the package. Instead, the structs are only created in the peer/rpcserver. This indicates that perhaps these structs shouldn't live in this package at all, or even a more aggressive refactoring is warranted.

Reply:
I moved them into `peer.go` where they are primarily used. More aggressive refactoring than this (for example creating a different pathway than that `interface{}` channel), I'd like to leave for another time. This PR is already a sub sub task.

Reply:
SGTM